### PR TITLE
Rf08 tc003 tc ivan misericordia 08

### PR DESCRIPTION
**Pull request by ivanMise**
RF08-TC-003 UI >Reporte de Inventario > Verificar Información de reporte de inventarios se muestre correctamente

image

image

image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class ListInventory extends BasePage{
+
+    @FindBy(className = "mat-button mat-primary")
+    private WebElement buttonAccept;
+
+    @FindBy(className = "mat-table")
+    private WebElement tableInventory;
+
+    @FindBy(xpath = ".//mat-card-header")
+    private WebElement imageReportHeader;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java]: rename webElement Ej.: headerImageReport

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    @FindBy(xpath = ".//mat-card-content/h1")
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java]: try tu use contains in xpath to find exactly the title

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private WebElement titleReportHeader;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java]: rename webElement Ej.: titleReport

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java [hunk 4]
```diff
@@ -47,9 +50,12 @@ public IncidentPage clickMenuIncident() {
         return new IncidentPage();
     }
 
-    public SubMenuEquipment selectEquipmentMenu() {
+    public List<BasePage> selectEquipmentMenu() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java]: the return should be atomic in every action, not is correct return a list

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 5]
```diff
@@ -202,10 +212,12 @@ public void validarQuePaginadorSeMuestraDebajoDeLaTablaListaDeCargo() throws Thr
...
     //End Positions
 
-    //BEGIN Equipment
+    //BEGIN Equipments
     @And("^seleccionar menu 'Equipamiento' en la pagina 'Menu Principal'$")
     public void menuEquipamientoEstaSeleccionado() throws Throwable {
-        menuEquipamiento = menu.selectEquipmentMenu();
+        List<BasePage> subMenusEquipments = menu.selectEquipmentMenu();
+        menuEquipamiento =(SubMenuEquipment) subMenusEquipments.get(0);
+        menuInventory = (SubMenuInventory) subMenusEquipments.get(1);
```
Comment (jdpapiero) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: the subMenuInventory gets after action of click in button equipment, so, just return menuEquipment in this case, review other examples of how use the interactions between menu events Ej.: Position, User, ProgramSSO.

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Feature: Check Inventory Report Header
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar cabecera de reporte de Inventario: Contenido e imagen
```
Comment (jdpapiero) [inline src/test/resources/features/equipments/checkHeaderReportInventory.feature]: in this case review you test case writed, is not the same.

**Review by jdpapiero — CHANGES_REQUESTED** [review]
review observations

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java (hunk 1 above)
Comment (ivanMise) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java]: done

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java (hunk 2 above)
Comment (ivanMise) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java]: done: @FindBy(xpath = ".//h1[contains(text(),'REPORTE DE ESTADO DE EQUIPAMIENTOS DE H.S.I')]")

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java (hunk 3 above)
Comment (ivanMise) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java]: done

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java (hunk 4 above)
Comment (ivanMise) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java]: done

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 5 above)
Comment (ivanMise) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: done

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature (hunk 6 above)
Comment (ivanMise) [reply src/test/resources/features/equipments/checkHeaderReportInventory.feature]: image

Comment (ivanMise) [reply src/test/resources/features/equipments/checkHeaderReportInventory.feature]: in this PR are two TCs the RF08-TC-001 and the RF08-TC-003

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java [hunk 7]
```diff
@@ -4,13 +4,17 @@
...
+import org.umssdiplo.automationv01.core.managepage.Equipment.ListInventory;
 import org.umssdiplo.automationv01.core.utils.CommonEvents;
 
 public class SubMenuEquipment extends BasePage {
 
     @FindBy(id = "smEquipament")
     private WebElement equipmentSubMenu;
 
+    @FindBy(id = "smInventory")
+    private WebElement inventorySubMenu;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java]: Rename web element ej. submenuInventory

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java [hunk 8]
```diff
@@ -4,13 +4,17 @@
 import org.openqa.selenium.support.FindBy;
 import org.umssdiplo.automationv01.core.managepage.BasePage;
 import org.umssdiplo.automationv01.core.managepage.Equipment.ListEquipment;
+import org.umssdiplo.automationv01.core.managepage.Equipment.ListInventory;
 import org.umssdiplo.automationv01.core.utils.CommonEvents;
 
 public class SubMenuEquipment extends BasePage {
 
     @FindBy(id = "smEquipament")
     private WebElement equipmentSubMenu;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java]: Rename ej.. submenuEquipment

**Review by jdpapiero — CHANGES_REQUESTED** [review]

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java (hunk 7 above)
Comment (ivanMise) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java]: done

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java (hunk 8 above)
Comment (ivanMise) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuEquipment.java]: done

**Review by jdpapiero — APPROVED** [review]
Approved

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature [hunk 9]
```diff
@@ -0,0 +1,12 @@
+Feature: Check Inventory Report Header
```
Comment (marbusjim) [inline src/test/resources/features/equipments/checkHeaderReportInventory.feature]: use spanish

File: src/test/resources/features/equipments/checkInventoryReport.feature [hunk 10]
```diff
@@ -0,0 +1,11 @@
+Feature: Check Inventory report
```
Comment (marbusjim) [inline src/test/resources/features/equipments/checkInventoryReport.feature]: use spanish

**Review by marbusjim — COMMENTED** [review]
with comments

**Review by marbusjim — APPROVED** [review]
approved

**Pull request merged**